### ISSUE TITLE
multiple threads support

### ISSUE THREAD

**Issue by bung87** [bung87/scorper#29]
use workers handle request.

**Comment by bung87** [comment]
I may finally turn the multithread model to `httpbeast`, in the meanwhile may not heavily relys on `chronos`

**Comment by bung87** [comment]
options:  
threads with shared ptrs: https://github.com/cheatfate/wanted
multiple server per threads bind on same address and port with event driven: httpbeast
use nginx as front with multiple processes

**Comment by shayanhabibi** [comment]
The issue with httpbeast however being that it doesn't support windows platform :(

**Comment by bung87** [comment]
we can use https://github.com/xflywind/httpx which use https://github.com/xflywind/ioselectors support windows platform, httpx is basically httpbeast with ioselectors. further more we can replace async macros with chronos's .

**Comment by shayanhabibi** [comment]
Performance with multithreading atm with httpx seems to [suffer on windows](https://github.com/xflywind/httpx/issues/2)

But then how big that impact would be using chronos would be interesting.

I would love to help; unfortunately macros isn't something I've had time to work with; same for networking hahahaha.

**Comment by bung87** [comment]
hmm, It's big like change scorper major version , need rewrite all stream handle from std socket to  chronos streamtransport , start server like httpbeast does . it's big I think.

**Comment by sdmcallister** [comment]
@bung87 

>I may finally turn the multithread model to httpbeast, in the meanwhile may not heavily relys on chronos

Seems like a big change. Would this significantly improve (what seems like already good) performance?

**Comment by bung87** [comment]
yes , it will significantly improve performance, It will not much relys on chronos only async macros I think, It will be a big change , I have not decide make changes in scorper repository or create new prjoect, it will be much like move current apis to httpbeast.
